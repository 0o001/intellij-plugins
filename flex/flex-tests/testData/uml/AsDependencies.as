package {
 var v__ = new Foo();

    class As<caret expected="">Dependencies {
        var f : Foo;
        var v : Vector.<Bar>;

        var t : Zz = new Zz();
        static var t = new Zz2();

        var we: Vector.<AsDependencies>;

    var f1 = new A8();
    var f2: Vector.<A8>;
    var f3: A8;

    var f5: Vector.<A9>;
    var f6: A9;

    var xx = A10.foo();
    var xx2 = A10.foo();

    function aa2(o: Oo, v123: Vector.<W>): Pp {
    }
    function aa(o: Oo, v123: Vector.<W>): Pp {
            var u: Abc = new Def();
            var u2: Rt;

            var v3 = new UI();
            var v4 = new UI();

            new Inner1();

            v = aa(new Inner2());

            var ttt = new Vector.<A1>();

        var x235: Vector.<A2>;
        var x234: A2;

            var me: AsDependencies;
            new AsDependencies();
            var v : NotShown;

        var x336 = new A3();
        var x335: Vector.<A3>;
        var x334: A3;

        var x436 = new A4();
        var x434: A4;
        var x435: Vector.<A4>;

        var static = A10.foo();
        new Foo(A10.foo());
    }
    include "AsDependencies_2.as"
}

class Foo {}
class Bar {}
class Zz {}
class Zz2{
    public function Zz2() {
    }
}
class Pp {}
class Oo {}
class W {}
class Abc {}
class Def {}
class Rt {}
class UI {}
class A1 {}
class A2 {}
class A3 {}
class A4 {}
class A5 {}
class A6 {}
class A7 {}
class A8 {}
class A9 {}
class NotShown {}
class A10{ public static function foo():void {} }
}

class Inner1 {}
class Inner2 {}
