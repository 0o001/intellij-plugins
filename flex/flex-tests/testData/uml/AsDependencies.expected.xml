<?xml version="1.0" encoding="UTF-8"?>
<umlModel origin="AsDependencies">
  <node id="A1" nodeName="Class A1" name="&lt;html&gt;&lt;b&gt;A1&lt;/b&gt;&lt;/html&gt;" />
  <node id="A10" nodeName="Class A10" name="&lt;html&gt;&lt;b&gt;A10&lt;/b&gt;&lt;/html&gt;">
    <element name="foo()" type="void" category="Methods" visibility="PUBLIC" />
  </node>
  <node id="A2" nodeName="Class A2" name="&lt;html&gt;&lt;b&gt;A2&lt;/b&gt;&lt;/html&gt;" />
  <node id="A3" nodeName="Class A3" name="&lt;html&gt;&lt;b&gt;A3&lt;/b&gt;&lt;/html&gt;" />
  <node id="A4" nodeName="Class A4" name="&lt;html&gt;&lt;b&gt;A4&lt;/b&gt;&lt;/html&gt;" />
  <node id="A5" nodeName="Class A5" name="&lt;html&gt;&lt;b&gt;A5&lt;/b&gt;&lt;/html&gt;" />
  <node id="A6" nodeName="Class A6" name="&lt;html&gt;&lt;b&gt;A6&lt;/b&gt;&lt;/html&gt;" />
  <node id="A7" nodeName="Class A7" name="&lt;html&gt;&lt;b&gt;A7&lt;/b&gt;&lt;/html&gt;" />
  <node id="A8" nodeName="Class A8" name="&lt;html&gt;&lt;b&gt;A8&lt;/b&gt;&lt;/html&gt;" />
  <node id="A9" nodeName="Class A9" name="&lt;html&gt;&lt;b&gt;A9&lt;/b&gt;&lt;/html&gt;" />
  <node id="Abc" nodeName="Class Abc" name="&lt;html&gt;&lt;b&gt;Abc&lt;/b&gt;&lt;/html&gt;" />
  <node id="AsDependencies" nodeName="Class AsDependencies" name="&lt;html&gt;&lt;b&gt;AsDependencies&lt;/b&gt;&lt;/html&gt;">
    <element name="i1" type="A5" visibility="PACKAGE_LOCAL" />
    <element name="i2" type="Vector.&lt;A6&gt;" visibility="PACKAGE_LOCAL" />
    <element name="i3" type="AsDependencies" visibility="PACKAGE_LOCAL" />
    <element name="f" type="Foo" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="v" type="Vector.&lt;Bar&gt;" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="t" type="Zz" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="t" type="*" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="we" type="Vector.&lt;AsDependencies&gt;" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="f1" type="*" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="f2" type="Vector.&lt;A8&gt;" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="f3" type="A8" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="f5" type="Vector.&lt;A9&gt;" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="f6" type="A9" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="xx" type="*" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="xx2" type="*" category="Fields" visibility="PACKAGE_LOCAL" />
    <element name="aa2(Oo, Vector.&lt;W&gt;)" type="Pp" category="Methods" visibility="PACKAGE_LOCAL" />
    <element name="aa(Oo, Vector.&lt;W&gt;)" type="Pp" category="Methods" visibility="PACKAGE_LOCAL" />
  </node>
  <node id="Bar" nodeName="Class Bar" name="&lt;html&gt;&lt;b&gt;Bar&lt;/b&gt;&lt;/html&gt;" />
  <node id="Def" nodeName="Class Def" name="&lt;html&gt;&lt;b&gt;Def&lt;/b&gt;&lt;/html&gt;" />
  <node id="Foo" nodeName="Class Foo" name="&lt;html&gt;&lt;b&gt;Foo&lt;/b&gt;&lt;/html&gt;" />
  <node id="Inner1" nodeName="Class Inner1" name="&lt;html&gt;&lt;b&gt;Inner1&lt;/b&gt;&lt;/html&gt;" />
  <node id="Inner2" nodeName="Class Inner2" name="&lt;html&gt;&lt;b&gt;Inner2&lt;/b&gt;&lt;/html&gt;" />
  <node id="Object" nodeName="Class Object" name="&lt;html&gt;&lt;b&gt;Object&lt;/b&gt;&lt;/html&gt;">
    <element name="constructor" type="*" category="Properties" visibility="PUBLIC" />
    <element name="length" type="Number" category="Properties" visibility="PUBLIC" />
    <element name="prototype" type="Object" category="Properties" visibility="PUBLIC" />
    <element name="toLocaleString()" type="Object" category="Methods" visibility="PUBLIC" />
    <element name="toSource()" type="Object" category="Methods" visibility="PUBLIC" />
    <element name="unwatch(String)" type="void" category="Methods" visibility="PUBLIC" />
    <element name="watch(String, Object)" type="void" category="Methods" visibility="PUBLIC" />
    <element name="hasOwnProperty(String)" type="Boolean" category="Methods" visibility="PUBLIC" />
    <element name="isPrototypeOf(Object)" type="Boolean" category="Methods" visibility="PUBLIC" />
    <element name="propertyIsEnumerable(String)" type="Boolean" category="Methods" visibility="PUBLIC" />
    <element name="toString()" type="String" category="Methods" visibility="PUBLIC" />
  </node>
  <node id="Oo" nodeName="Class Oo" name="&lt;html&gt;&lt;b&gt;Oo&lt;/b&gt;&lt;/html&gt;" />
  <node id="Pp" nodeName="Class Pp" name="&lt;html&gt;&lt;b&gt;Pp&lt;/b&gt;&lt;/html&gt;" />
  <node id="Rt" nodeName="Class Rt" name="&lt;html&gt;&lt;b&gt;Rt&lt;/b&gt;&lt;/html&gt;" />
  <node id="UI" nodeName="Class UI" name="&lt;html&gt;&lt;b&gt;UI&lt;/b&gt;&lt;/html&gt;" />
  <node id="W" nodeName="Class W" name="&lt;html&gt;&lt;b&gt;W&lt;/b&gt;&lt;/html&gt;" />
  <node id="Zz" nodeName="Class Zz" name="&lt;html&gt;&lt;b&gt;Zz&lt;/b&gt;&lt;/html&gt;" />
  <node id="Zz2" nodeName="Class Zz2" name="&lt;html&gt;&lt;b&gt;Zz2&lt;/b&gt;&lt;/html&gt;">
    <element name="Zz2()" category="Constructors" visibility="PUBLIC" />
  </node>
  <edge from="A1" to="Object" idElement="A1" relship="GENERALIZATION" />
  <edge from="A10" to="Object" idElement="A10" relship="GENERALIZATION" />
  <edge from="A2" to="Object" idElement="A2" relship="GENERALIZATION" />
  <edge from="A3" to="Object" idElement="A3" relship="GENERALIZATION" />
  <edge from="A4" to="Object" idElement="A4" relship="GENERALIZATION" />
  <edge from="A5" to="Object" idElement="A5" relship="GENERALIZATION" />
  <edge from="A6" to="Object" idElement="A6" relship="GENERALIZATION" />
  <edge from="A7" to="Object" idElement="A7" relship="GENERALIZATION" />
  <edge from="A8" to="Object" idElement="A8" relship="GENERALIZATION" />
  <edge from="A9" to="Object" idElement="A9" relship="GENERALIZATION" />
  <edge from="Abc" to="Object" idElement="Abc" relship="GENERALIZATION" />
  <edge from="AsDependencies" to="A1" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="A10" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="A10" idElement="AsDependencies" relship="DEPENDENCY" relshipLabel="xx" />
  <edge from="AsDependencies" to="A10" idElement="AsDependencies" relship="DEPENDENCY" relshipLabel="xx2" />
  <edge from="AsDependencies" to="A2" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="A3" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="A3" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="A4" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="A4" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="A5" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="i1" />
  <edge from="AsDependencies" to="A6" idElement="AsDependencies" relship="ONE_TO_MANY" relshipLabel="i2" />
  <edge from="AsDependencies" to="A8" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="A8" idElement="AsDependencies" relship="ONE_TO_MANY" relshipLabel="f2" />
  <edge from="AsDependencies" to="A8" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="f3" />
  <edge from="AsDependencies" to="A9" idElement="AsDependencies" relship="ONE_TO_MANY" relshipLabel="f5" />
  <edge from="AsDependencies" to="A9" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="f6" />
  <edge from="AsDependencies" to="Abc" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="AsDependencies" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="AsDependencies" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="AsDependencies" idElement="AsDependencies" relship="ONE_TO_MANY" relshipLabel="we" />
  <edge from="AsDependencies" to="AsDependencies" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="i3" />
  <edge from="AsDependencies" to="Bar" idElement="AsDependencies" relship="ONE_TO_MANY" relshipLabel="v" />
  <edge from="AsDependencies" to="Def" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="Foo" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="Foo" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="f" />
  <edge from="AsDependencies" to="Inner1" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="Inner2" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="Object" idElement="AsDependencies" relship="GENERALIZATION" />
  <edge from="AsDependencies" to="Oo" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="Pp" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="Rt" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="UI" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="W" idElement="AsDependencies" relship="DEPENDENCY" />
  <edge from="AsDependencies" to="Zz" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="AsDependencies" to="Zz" idElement="AsDependencies" relship="ONE_TO_ONE" relshipLabel="t" />
  <edge from="AsDependencies" to="Zz2" idElement="AsDependencies" relship="CREATE" relshipLabel="&lt;html&gt;&amp;laquo;create&amp;raquo;&lt;/html&gt;" />
  <edge from="Bar" to="Object" idElement="Bar" relship="GENERALIZATION" />
  <edge from="Def" to="Object" idElement="Def" relship="GENERALIZATION" />
  <edge from="Foo" to="Object" idElement="Foo" relship="GENERALIZATION" />
  <edge from="Inner1" to="Object" idElement="Inner1" relship="GENERALIZATION" />
  <edge from="Inner2" to="Object" idElement="Inner2" relship="GENERALIZATION" />
  <edge from="Oo" to="Object" idElement="Oo" relship="GENERALIZATION" />
  <edge from="Pp" to="Object" idElement="Pp" relship="GENERALIZATION" />
  <edge from="Rt" to="Object" idElement="Rt" relship="GENERALIZATION" />
  <edge from="UI" to="Object" idElement="UI" relship="GENERALIZATION" />
  <edge from="W" to="Object" idElement="W" relship="GENERALIZATION" />
  <edge from="Zz" to="Object" idElement="Zz" relship="GENERALIZATION" />
  <edge from="Zz2" to="Object" idElement="Zz2" relship="GENERALIZATION" />
</umlModel>

